### PR TITLE
Adjustments to make mypyc working in adviser

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Related: https://github.com/thoth-station/adviser/pull/2173

## This introduces a breaking change

- [x] No

### CODE REVIEW

**Comment by goern** [comment]
ping

File: thoth/python/package_version.py [hunk 1]
```diff
@@ -393,13 +393,31 @@ def to_tuple(self) -> Tuple[str, str, Optional[str]]:
 
         return self._package_tuple
 
+    def to_strict_tuple(self) -> Tuple[str, str, str]:
+        """Return a tuple representing this Python package, packages with index assigned."""
+        assert self.index is not None
+
+        if self._package_tuple:
```
Comment (KPostOffice) [inline thoth/python/package_version.py:400]:
```suggestion
        if self._package_tuple is None:
```
I think this should be a check for None

File: thoth/python/package_version.py [hunk 2]
```diff
@@ -393,13 +393,31 @@ def to_tuple(self) -> Tuple[str, str, Optional[str]]:
...
         if not self._package_tuple_locked:
             self._package_tuple_locked = self.name, self.locked_version, self.index.url if self.index else None
 
         return self._package_tuple_locked
 
+    def to_strict_tuple_locked(self) -> Tuple[str, str, str]:
+        """Return a tuple representing this Python package - locked packages with index assigned."""
+        assert self.index is not None
+
+        if self._package_tuple:
```
Comment (KPostOffice) [inline thoth/python/package_version.py:416]:
```suggestion
        if self._package_tuple is None:
```
Same here.

**Comment by fridex** [comment]
I was not successful with mypyc in the adviser implementation. It looks like mypyc is not stable enough to consider it. I will close this PR. Feel free to raise any comments or suggestions.

/close

**Pull request closed without merging**